ilastik/experimental/api/_pipelines.py

```diff
@@ -0,0 +1,256 @@
+        self._reorder_raw_op = OpReorderAxes(graph=self._graph, AxisOrder="txyzc")
+        self._reorder_seg_op = OpReorderAxes(graph=self._graph, AxisOrder="txyzc")
+        self._obj_extraction_op.RawImage.connect(self._reorder_raw_op.Output)
+        self._obj_extraction_op.BinaryImage.connect(self._reorder_seg_op.Output)
+
+    def get_object_probabilities(self, raw_data, segmentation_image):
+        return self._process(raw_data, segmentation_image, output_slot=self._stack_prob_channels.Output)
+
+    def _process(self, raw_data, segmentation_image, output_slot):
+        raw_data = convert_to_vigra(raw_data)
+        input_axistags = "".join(ax.key for ax in raw_data.axistags)
+        segmentation_image = convert_to_vigra(segmentation_image)
+        self._check_data(raw_data)
+        self._reorder_raw_op.Input.setValue(raw_data)
+        self._reorder_seg_op.Input.setValue(segmentation_image)
+        processed_data = output_slot.value[...]
+        return _reorder_output(processed_data, "".join(output_slot.meta.axistags.keys()), input_axistags)
+
+
+class _ObjectClassificationFromPredictionPipelineImpl(
+    _ObjectClassificationPipelineBase, ObjectClassificationFromPredictionPipeline
+):
+    def __init__(self, project: parser.ObjectClassificationProjectBase):
+        super().__init__(project)
+        self._reorder_raw_op = OpReorderAxes(graph=self._graph, AxisOrder="txyzc")
+        self._reorder_pred_op = OpReorderAxes(graph=self._graph, AxisOrder="txyzc")
+
+        self._normalize_probmaps = OpPixelOperator(graph=self._graph)
+        self._normalize_probmaps.Input.connect(self._reorder_pred_op.Output)
+
+        thresh_settings = project.thresholding_settings
+        self._thresholding_op = OpThresholdTwoLevels(graph=self._graph)
+        self._thresholding_op.RawInput.connect(self._reorder_raw_op.Output)
+        self._thresholding_op.MinSize.setValue(thresh_settings.min_size)
+        self._thresholding_op.MaxSize.setValue(thresh_settings.max_size)
+        self._thresholding_op.HighThreshold.setValue(thresh_settings.high_threshold)
+        self._thresholding_op.LowThreshold.setValue(thresh_settings.low_threshold)
+        self._thresholding_op.SmootherSigma.setValue(thresh_settings.smoother_sigmas)
+        self._thresholding_op.Channel.setValue(thresh_settings.channel)
+        self._thresholding_op.CoreChannel.setValue(thresh_settings.core_channel)
+        self._thresholding_op.CurOperator.setValue(thresh_settings.method)
+        self._thresholding_op.InputImage.connect(self._normalize_probmaps.Output)
+
+        self._obj_extraction_op.RawImage.connect(self._reorder_raw_op.Output)
+        self._obj_extraction_op.BinaryImage.connect(self._thresholding_op.Output)
+
+    def get_object_probabilities(self, raw_data, prediction_maps):
+        return self._process(raw_data, prediction_maps, output_slot=self._stack_prob_channels.Output)
+
+    def _setup_normalization_function(self, prediction_maps: numpy.ndarray):
```

Comment:
This makes the pipeline not thread-safe (but then again, maybe more stuff inside classic ilastik already makes it not thread-safe when used as a library like this anyway). Are we already settled in the concurrency limitations we want to impose on the consumer of the API?